pkg/kubelet/kubelet_pods.go

```diff
@@ -639,12 +640,20 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *v1.Pod, container *v1.Container
 					invalidKeys = append(invalidKeys, k)
 					continue
 				}
+				_, ok := tmpEnv[k]
+				if ok {
+					duplicatedKeys = append(duplicatedKeys, k)
+				}
 				tmpEnv[k] = string(v)
 			}
 			if len(invalidKeys) > 0 {
 				sort.Strings(invalidKeys)
 				kl.recorder.Eventf(pod, v1.EventTypeWarning, "InvalidEnvironmentVariableNames", "Keys [%s] from the EnvFrom secret %s/%s were skipped since they are considered invalid environment variable names.", strings.Join(invalidKeys, ", "), pod.Namespace, name)
 			}
+			if len(duplicatedKeys) > 0 {
+				sort.Strings(duplicatedKeys)
+				kl.recorder.Eventf(pod, v1.EventTypeWarning, "DuplicatedEnvironmentVariableNames", "Keys [%s] from the EnvFrom secret %s/%s were already defined by other configMap or secret. Use %s/%s.", strings.Join(duplicatedKeys, ", "), pod.Namespace, name, pod.Namespace, name)
```

Comment:
I'm not sure I understand the purpose of the `Use %s/%s` instruction. Are you telling the user to use the already defined key? What if the keys have separate values?

Reply:
Also, could the keys only be from the `EnvFrom secret`? Couldn't they also be from a ConfigMap?

Reply:
> Also, could the keys only be from the EnvFrom secret? Couldn't they also be from a ConfigMap?

I missed the `switch` . I need to fix to evaluate both ConfigMap and Secret at the same time.
I'll fix it.

Reply:
> I'm not sure I understand the purpose of the Use %s/%s instruction. Are you telling the user to use the already defined key? What if the keys have separate values?

In this inplementation of `makeEnvironmentVariables`, it has a map `tmpEnv` and all of envs are insert into it.
Also, the map `tmpEnv` keys are using as env key.
If the env key already that is defined from other file is inserted, the value will override.
Then, pod will get the value from the file last evaluated.
Thus, I think it need what resource's env are used as a result.

Reply:
Gotcha - please ping me when the code is updated to address the `switch` :)

I still am not sure that I understand the point of the `Use %s/%s.` Thoughts on if the log message would be more readable without it?

Reply:
@mattjmcnaughton 
bd5a1f3
I updated the code:  
- Moved `duplicatedKeys` to top of func to use for both ConfigMap and Secret.
- Add consideration for all declare means: 
  - env array (include both plain and valueFrom)
  - envFrom (ConfigMap and Secret)

and, some tests add.
I missed seeing below comment from you, I will check and reply.